### PR TITLE
Add conversational and text-to-image pipelines to Interface.load

### LINKED ISSUE #2982: Add conversational pipeline to Interface.load
- [X] I have searched to see if a similar issue already exists.

When trying to use the HuggingFace hosted Chatbot-models the Interface.load fails with the error:
`ValueError: Unsupported pipeline type: conversational`

The hosted models on HuggingFace are really useful for testing out models without having to download them. And also makes it possible to use large models while developing on a laptop.

So it would be very useful to be able to use conversational models via the HTTP load functionality.

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Fixes #2982 
Fixes: #2956 

Added unit tests but you can also test with `gr.Interface.load("models/microsoft/DialoGPT-medium")` (conversational) and `gr.Interface.load("models/nlpconnect/vit-gpt2-image-captioning")` (text-to-image)

## Conversational demo
chatbot_load

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
-No changes to highlight.
+
+### Extended support for Interface.load! 🏗️
+
+You can now load `text-to-image` and `conversational` pipelines from the hub!
+
+### text-to-image Demo
+```python
+io = gr.Interface.load("models/nlpconnect/vit-gpt2-image-captioning",
+                       api_key="<your-api-key>")
```
Comment (abidlabs) [inline CHANGELOG.md:12]: Suggestion to make it clearer that api keys are optional:
```suggestion
                       api_key="<optional-api-key>")
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+io.launch()
+```
+<img width="1087" alt="image" src="https://user-images.githubusercontent.com/41651716/213260197-dc5d80b4-6e50-4b3a-a764-94980930ac38.png">
+
+
+
+### conversational Demo
+```python
+chatbot = gr.Interface.load("models/microsoft/DialoGPT-medium",
+                           api_key="<your-api-key>")
```
Comment (abidlabs) [inline CHANGELOG.md:22]: Suggestion to make it clearer that api keys are optional:
```suggestion
                           api_key="<optional-api-key>")
```

**Comment by abidlabs** [comment]
Tested this with a few models and it works great @freddyaboulton! I think we need to fix the messaging on the changelog since you've added an image-to-text not text-to-image, but otherwise this is awesome

**Review by abidlabs — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,31 @@
 # Upcoming Release 
 
 ## New Features:
-No changes to highlight.
+
+### Extended support for Interface.load! 🏗️
+
+You can now load `text-to-image` and `conversational` pipelines from the hub!
+
+### text-to-image Demo
```
[the remark this answers is not in the document]
Comment (freddyaboulton) [reply CHANGELOG.md:9]: Omg pretty embarrassing 🙈

**Pull request merged**